agent/engine/docker_image_manager.go

```diff
@@ -72,6 +74,19 @@ func NewImageManager(cfg *config.Config, client DockerClient, state dockerstate.
 	}
 }
 
+// WhitelistImage add the image into whitelists that won't be cleaned up by agent
+// This is safe without a lock for now, as it's only called when agent start
```

Comment:
Please add a lock or change the function name to include the word unsafe. I'd rather not make assertions like "this is safe for now".